tensorflow/python/ops/gradients_test.py

```diff
@@ -419,6 +419,36 @@ def testHessianVectorProduct(self):
       self.assertAllClose(hess_v_value, hess_v_actual)
 
 
+class HessianTest(test_util.TensorFlowTestCase):
+
+  def testHessian1D(self):
+    # Manually compute the Hessian explicitly for a low-dimensional problem
+    # and check that `hessian` matches. Specifically, the Hessian of 
+    # f(x) = x^T A x is H = A + A^T.
+    m = 4
+    rng = np.random.RandomState([1, 2, 3])
+    mat_value = rng.randn(m, m).astype("float32")
+    x_value = rng.randn(m).astype("float32")
+    hess_value = mat_value + mat_value.T
+    for use_gpu in [False, True]:
```

Comment:
here and below, you can avoid the loop and just do: use_gpu=True.  It will try to use gpu when available and use CPU otherwise.